appservice/src/createAppService/SiteOSStep.ts

```diff
@@ -20,6 +21,18 @@ export class SiteOSStep extends AzureWizardPromptStep<IAppServiceWizardContext>
             wizardContext.newSiteOS = (await ext.ui.showQuickPick(picks, { placeHolder: localize('selectOS', 'Select an OS.') })).data;
         }
 
+        let options: {} = {};
+        if (wizardContext.newSiteOS === 'linux') {
+            if (wizardContext.newSiteKind === AppKind.functionapp) {
+                options = { linuxDynamicWorkersEnabled: true };
+            } else {
+                options = { linuxWorkersEnabled: true };
+            }
+        }
+        // Overwrite the generic 'locationsTask' with a list of locations specific to provider "Microsoft.Web"
+        const client: WebSiteManagementClient = createAzureClient(wizardContext, WebSiteManagementClient);
+        wizardContext.locationsTask = client.listGeoRegions(options);
```

Comment:
It's kinda weird to me that we're setting the locations in the OS code (though I get that it's dependant on it being Linux)